Static attention IO manager

sxu opened the pull request:
Summary: Helper class that manages I/O tensors for KV caches, attention masks, and RoPE frequencies. Data types of these tensors are templated to support quantized data.

Differential Revision: D69624077

iseeyuan reviewed (approved):
LGTM. Please add runner file in the following PR to cover the tests.

sxu committed:
Static attention IO manager (#8486)

Summary:

Helper class that manages I/O tensors for KV caches, attention masks, and RoPE frequencies. Data types of these tensors are templated to support quantized data.

Reviewed By: iseeyuan

Differential Revision: D69624077
